src/app/shared/session/session.query.ts

```diff
@@ -41,23 +42,25 @@ export class SessionQuery extends Query<SessionState> {
   isService$: Observable<boolean> = this.entryType$.pipe(map((entryType) => entryType === EntryType.Service));
   isTool$: Observable<boolean> = this.entryType$.pipe(map((entryType) => entryType === EntryType.Tool));
   gitHubAppInstallationLink$: Observable<string> = this.entryType$.pipe(
-    map((entryType: EntryType) => (entryType ? this.generateGitHubAppInstallationUrl(entryType) : null))
+    map((entryType: EntryType) => (entryType ? this.generateGitHubAppInstallationUrl(this.route.url) : null))
   );
   loadingDialog$: Observable<boolean> = this.select((session) => session.loadingDialog);
-  constructor(protected store: SessionStore) {
+  constructor(protected store: SessionStore, private route: Router) {
     super(store);
   }
 
   /**
    * Generate the general GitHub App installation URL
    *
-   * @param {EntryType} entryType  To determine which page the user currently is on
+   * @param {string} redirectUrl  The page to redirect to after installation is complete
    * @returns {string}
    * @memberof WorkflowQuery
    */
-  generateGitHubAppInstallationUrl(entryType: EntryType): string {
+  generateGitHubAppInstallationUrl(redirectUrl: string): string {
```

Comment:
Nit, even though the cause of this is Angular misusing the term `url`, or me not understanding what an url is, I think the parameter is really a path, not an URL (it's `/my-workflows/github.com/A/l`, per your tests, not `https://dockstore.org/my-workflows/github.com/A/l`).

Optionally update the doc and/or parameter name. Or leave it and be consistent with Angular terminology. :)

I only looked into it because I thought it was a fully qualified URL, and was checking if the `navigateByUrl` above would cause an unnecessary reload of the page (it doesn't).